barretenberg/cpp/src/barretenberg/eccvm/eccvm_flavor.hpp

```diff
@@ -1506,6 +1491,22 @@ class ECCVMFlavor {
             ASSERT(NativeTranscript::proof_data.size() == old_proof_length);
         }
     };
+
+    /**
+     * @brief When evaluating the sumcheck protocol - can we skip evaluation of all relations for a given row?
+     *
+     * @details When used in ClientIVC, the ECCVM has a large fixed size, which is often not fully utilized.
+     *          If a row is completely empty, the values of z_perm and z_perm_shift will match,
+     *          we can use this as a proxy to determine if we can skip Sumcheck::compute_univariate
+     **/
+    template <typename ProverPolynomialsOrPartiallyEvaluatedMultivariates, typename EdgeType>
+    static bool skip_entire_row([[maybe_unused]] const ProverPolynomialsOrPartiallyEvaluatedMultivariates& polynomials,
```

Comment:
Question: why not use the already existing `skippable` system? It's a property (a method) of a given Relation, however, not of the flavor, and I understand you might want something more global, but maybe `skip_entire_row` would be equivalent to "all relations have [skip()](https://github.com/AztecProtocol/aztec-packages/blob/2b218153e3ecbec6096436ee227ee082f67f9823/barretenberg/cpp/src/barretenberg/relations/ecc_op_queue_relation.hpp#L27) defined and they all return true".

Reply:
I want to determine ahead-of-time which rows can be skipped, so that when we multithread we can ensure each thread has an equal amount of work to perform (instead of e.g. 1 thread computing the 128 rows that are not skipped and every other thread computing 128 skipped rows)

I could do that by iterating over every relation for every row, but for the VMs the skip condition is the same for every row so it would be a lot of redundant work

Reply:
> but for the VMs the skip condition is the same for every row so it would be a lot of redundant work

Might be true, how many relations do you have? I guess it's the tradeoff between further complicating sumcheck or paying some cost by checking all of the relations' `skip`. Wouldn't be able to tell without the data :) in any case I don't own this code so it's up to the crypto team